### PR TITLE
Reorganize the high-scale tuning section

### PR DESCRIPTION
**Pull request by moio**
Currently we have some tuning settings in the Scalability part, others (DB, Apache connections) are in different points in the manual.

As highlighted by our users we should create a new section/chapter about tuning and group all content there in a form which is more easily consumable.

Part of https://github.com/SUSE/spacewalk/issues/5217

### CODE REVIEW

**Comment by moio** [comment]
Documentation Team: I opened this as a Draft as I continue adding parameters. My intention would be to pass it over to you as soon as the content is there, for you to handle language, style, consistency, etc.

If there's any glaring error just tell me so that I incorporate the fix already!

File: modules/salt/pages/salt-big-scale.adoc [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+
+The number of distinct OS versions that managed clients have installed, differentiated by family (SUSE Linux Enterprise-based, openSUSE-based, Red Hat-based or Ubuntu-based). Storage and computing requirements are different in each case.
+
+===== User count
+
+The expected maximum amount of concurrent users interacting with the Web UI plus the number of programs simultaneously using the XMLRPC API (including `spacecmd`, `spacewalk-clone-by-date`, etc.).
+
+==== Parameters
+
+===== `MaxClients`
```
Comment (admd) [inline modules/salt/pages/salt-big-scale.adoc]: Nitpick: Maybe better to mention it explicitly that `MaxClients` has been renamed to `MaxRequestWorkers`

**Comment by moio** [comment]
@Loquacity although I still have to make really-really sure I didn't forget any parameter around (that's something I will check tomorrow) - it should be a matter of a really few lines. The structure and most of the content is ready for review FMPOV.

I understand this has grown to a rather big one, with refactorings/moving of sections and lots of new material - still I hope this is digestible for Documentation people at this point!

Feel free to ask any question and just make edits directly if any aspect is obvious to you (in particular regarding asciidoc formatting).

@admd is also taking a look on the content side.

**Comment by Loquacity** [comment]
I definitely want to do some editing on the content, but it's probably easiest to merge this first, then I can do a line edit in a separate PR.

**Comment by admd** [comment]
I am merging this PR as @Loquacity suggested. Further changes could be part of another PR later.

File: modules/salt/pages/salt-big-scale.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (moio) [reply modules/salt/pages/salt-big-scale.adoc]: Fixed

File: modules/salt/pages/salt-big-scale.adoc [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+* Description: the maximum number of http requests served simultaneously by Apache httpd. Proxies, Web UI and XMLRPC API clients each consume one. Requests exceeding the parameter will be queued and might result in timeouts.
+* Tune when: user count and Proxy count increase significantly and the following line appears in `/var/log/apache2/error_log`:
+----
+[...] [mpm_prefork:error] [pid ...] AH00161: server reached MaxRequestWorkers setting, consider raising the MaxRequestWorkers setting
+----
+* Value guideline: 150 (default) - 500
+* Location: `/etc/apache2/server-tuning.conf`
+* After changing immediately change `ServerLimit` and also check `maxThreads` for possible adjustment
```
[the remark this answers is not in the document]
Comment (moio) [reply modules/salt/pages/salt-big-scale.adoc]: It's actually completely wrong, removed

File: modules/salt/pages/salt-big-scale.adoc [hunk 3]
```diff
@@ -320,6 +340,20 @@ The expected maximum amount of concurrent users interacting with the Web UI plus
 * Value guideline: 20 (default) - 200
 * Location: `/etc/rhn/rhn.conf`
 * After changing, also check for adjustment: `hibernate.c3p0.max_size`, `thread_pool`
+* More information on this parameter available at: http://www.quartz-scheduler.org/documentation/2.4.0-SNAPSHOT/configuration.html
+
+==== `org.quartz.scheduler.idleWaitTime`
+* Description: cycle time for Taskomatic. Decreasing this value lowers Taskomatic's latency.
+* Tune when: client count is in the thousands
+* Value guideline: 1000 - 5000 (default). Note that the value is expressed in milliseconds
```
[the remark this answers is not in the document]
Comment (moio) [reply modules/salt/pages/salt-big-scale.adoc]: Well, I tried it (in this context: https://github.com/SUSE/spacewalk/wiki/Patching-2.4k-minions-in-AWS---20171113) and I noticed no probems at all at least in the Manager context. Do you think it is worth it to add an explicit "counter-note"?

File: modules/salt/pages/salt-big-scale.adoc [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+* More information on this parameter available at: https://httpd.apache.org/docs/2.4/en/mod/mpm_common.html#serverlimit
+
+==== `maxThreads`
+* Description: the number of Tomcat threads dedicated to serving HTTP requests.
+* Tune when: `MaxClients` changes. `maxThreads` must always be equal or greater than `MaxClients`
+* Value guideline: equal to `MaxClients`
+* Location: `/etc/tomcat/server.xml`
+* More information on this parameter available at: https://tomcat.apache.org/tomcat-9.0-doc/config/http.html
+
+==== Tomcat's `-Xmx`
```
[the remark this answers is not in the document]
Comment (moio) [reply modules/salt/pages/salt-big-scale.adoc]: Hmm, my problem is that we never really tested it. Will do that at first occasion and eventually add it, thanks!

**Comment by moio** [comment]
> * I personally would have preferred grouping of these parameters based on the components or which configuration file parameter belongs to. If not then even alphabetical order would be ok.

Done.

> * I would prefer to have these parameters as links so navigation can be better when referring parameters from each other.

Now man, I must admit I hated you for a brief moment when I read this. But after rolling up my regexp sleeves I have to thank you so much, the document is so much easier to navigate now.

My only regret now is not being able to find a decent way to make the graph nodes clickable!!!

**Comment by moio** [comment]
>     3\. I would love to have a table with all of these parameters with their `default` values, it could even help us when fixing some L3 just to get a quick overview to see if the customer have some reasonable values in their setup.
> 
> We could even add a plugin in our supportconfig tool which just list such parameters in a single file.

I agree but this is nothing that should really be for end users, which the manuals are for.

We can easily extract the list of defaults or even create better tooling to help debugging, sure.

**Pull request merged**